src/webserver/DOSGuard.h

```diff
@@ -84,17 +415,65 @@ class BasicDOSGuard : public BaseDOSGuard
         sweepHandler.setup(this);
     }
 
+    /**
+     * @brief Checks if IP address is malformed
+     *
+     * @param ip IP address to check
+     */
+    [[nodiscard]] bool
+    isValidIPMixed(std::string const& ip) const
+    {
+        try
+        {
+            boost::asio::ip::address::from_string(ip);
+            return true;
+        }
+        catch (const std::exception& e)
+        {
+            return false;
+        }
+    }
+
     /**
      * @brief Check whether an ip address is in the whitelist or not
      *
-     * @param ip The ip address to check
+     * @param ip The ip address to check (raw IP, NO CIDR)
      * @return true
-     * @return false
+     * @return falseconvertedIPsToStrings
      */
     [[nodiscard]] bool
     isWhiteListed(std::string const& ip) const noexcept
     {
-        return whitelist_.contains(ip);
+        // CASE #1: Check if IP is CIDR. Treat as malformed input
+        if (ip.find("/") != std::string::npos)
```

Comment:
A bit redundant in logic. The IP is from the connection, we can be sure the IP is valid, right?